editor/src/components/canvas/controls/observer-hooks.ts

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import { ElementPath } from '../../../core/shared/project-file-types'
+import { fastForEach } from '../../../core/shared/utils'
+import * as EP from '../../../core/shared/element-path'
+import { CanvasRectangle, windowPoint } from '../../../core/shared/math-utils'
+import { windowToCanvasCoordinatesGlobal } from '../dom-lookup'
+
+function attachObserverToScrollingElement(
+  observerRef: React.MutableRefObject<MutationObserver | null | undefined>,
+) {
+  // TODO FIND SOMETHING NICER
+  // TODO use HTMLElement that is changing attributes on scroll (instead of cssvar) specifically for these controls
+  const thisElementRerendersOnScroll = document.getElementById('node-connectors')
```

Comment:
the original problem it was trying to solve is that we wanted to re-run the observer callback on canvas scroll. When we scroll the canvas, the MutationObserver for elements on the canvas does not fire, because the scroll is not mutating those elements, only changing their parent offset. 

but @enidemi we should make sure this is still necessary!

it might not be necessary anymore, because the first prototype implementation relied on this so that the controls would follow the canvas scroll, but we stopped doing that and the controls use the canvas offset css vars now.

Reply:
that's true, we no longer need this function